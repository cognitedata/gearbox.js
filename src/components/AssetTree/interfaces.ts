import { Asset } from '@cognite/sdk';
import { AutoPagingToArrayOptions } from '@cognite/sdk/dist/src/autoPagination';
import { AnyIfEmpty } from '../../interfaces';

export interface AssetTreeStyles {
  list?: React.CSSProperties;
}

export interface OnSelectAssetTreeParams {
  key: number | string;
  title: string;
  isLeaf?: boolean;
  node?: Asset;
}

export interface AssetTreeProps {
  /**
   * Modify the display name of an asset
   */
  displayName?: (asset: Asset) => string;
  /**
   * Triggers when a node is selected
   */
  onSelect?: (onSelect: OnSelectAssetTreeParams) => void;
  /**
   * To display a tree with provided assets as root nodes
   */
  assetIds?: number[];
  /**
   * Show loading animation when fetching original list
   */
  showLoading?: boolean;
  /**
   * List of node ids to be expanded by default
   */
  defaultExpandedKeys?: number[];
  /**
   * Object that defines inline CSS styles for
   * inner elements of the component.
   */
  styles?: AssetTreeStyles;
  /**
   * @ignore
   */
  theme?: AnyIfEmpty<{}>;

  autoPagingToArrayOptions?: AutoPagingToArrayOptions;
}
