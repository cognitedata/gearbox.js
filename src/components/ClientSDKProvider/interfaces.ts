import { CogniteClient } from '@cognite/sdk';
import React from 'react';

export interface ClientSDKProviderProps {
  client: CogniteClient;
  children: React.ReactNode;
}
