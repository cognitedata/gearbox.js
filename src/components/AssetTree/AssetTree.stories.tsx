import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import React from 'react';
import AssetTree from 'components/AssetTree/AssetTree';
import { ASSET_ZERO_DEPTH_ARRAY, ASSET_LIST_CHILD } from 'mocks/assetsList';

const returnLoadData = () => {
  return ASSET_LIST_CHILD;
};

const zeroChild = ASSET_LIST_CHILD.findIndex(asset => asset.depth === 0);

storiesOf('AssetTree', module)
  .add('All data on mount', () => <AssetTree assets={ASSET_LIST_CHILD} />)
  .add('Default expanded node', () => (
    <AssetTree
      assets={ASSET_LIST_CHILD}
      defaultExpandedKeys={[String(ASSET_LIST_CHILD[zeroChild].id)]}
    />
  ))
  .add(
    'Add data onLoad',
    () => (
      <AssetTree assets={ASSET_ZERO_DEPTH_ARRAY} loadData={returnLoadData} />
    ),
    {
      info: {
        text:
          'loadData Should return a list of assets. Typically a call to listDescendants.',
      },
    }
  )
  .add(
    'Click item in tree',
    () => (
      <AssetTree
        assets={ASSET_ZERO_DEPTH_ARRAY}
        loadData={returnLoadData}
        onSelect={e => action('onSelect')(e)}
      />
    ),
    {
      info: {
        text: 'onSelect returns the selected object (see action logger).',
      },
    }
  );
