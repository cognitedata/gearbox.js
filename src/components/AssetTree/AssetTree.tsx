import { Asset, CogniteClient } from '@cognite/sdk';
import { Spin, Tree } from 'antd';
import { AntTreeNode, AntTreeNodeProps } from 'antd/lib/tree';
import { isEqual } from 'lodash';
import React, { Component } from 'react';
import styled from 'styled-components';
import {
  ERROR_API_UNEXPECTED_RESULTS,
  ERROR_NO_SDK_CLIENT,
} from '../../constants/errorMessages';
import { ClientSDKProxyContext } from '../../context/clientSDKProxyContext';
import { withDefaultTheme } from '../../hoc';
import { defaultTheme } from '../../theme/defaultTheme';
import {
  applyThemeFontFamily,
  applyThemeFontSize,
  applyThemeListHighlight,
} from '../../utils/theme';
import { AssetTreeProps } from './interfaces';

const { TreeNode } = Tree;

interface ExpandedKeysMap {
  [key: number]: true;
}

interface AssetTreeNodeMap {
  [id: string]: AssetTreeNode;
}

interface AssetTreeState {
  loading: boolean;
  loadedKeys: string[];
  rootAssetNodes: number[];
  assetNodesMap: AssetTreeNodeMap;
  expandedKeys: ExpandedKeysMap;
}

interface AutoPagingToArrayOptions {
  limit?: number;
}

interface AssetTreeNode {
  asset: Asset;
  children?: number[];
  isLeaf: boolean;
}

class AssetTree extends Component<AssetTreeProps, AssetTreeState> {
  static contextType = ClientSDKProxyContext;
  static defaultProps = {
    theme: { ...defaultTheme },
  };

  static convertAssetsToNodeMap(assets: Asset[]): AssetTreeNodeMap {
    const nodes: AssetTreeNodeMap = {};
    assets.map(AssetTree.convertToNode).forEach(node => {
      nodes[node.asset.id] = node;
    });
    return nodes;
  }

  static convertToNode(asset: Asset): AssetTreeNode {
    const isLeaf = asset.aggregates ? !asset.aggregates.childCount : false;
    return { asset, isLeaf };
  }
  // TODO path?
  static toKeys(path: number[], initial = {}): ExpandedKeysMap {
    return path.reduce((acc, i) => ({ ...acc, [i]: true }), initial);
  }
  context!: React.ContextType<typeof ClientSDKProxyContext>;
  client!: CogniteClient;

  autoPagingToArrayOptions: AutoPagingToArrayOptions = this.props
    .autoPagingToArrayOptions
    ? this.props.autoPagingToArrayOptions
    : { limit: Infinity };

  constructor(props: AssetTreeProps) {
    super(props);
    const { defaultExpandedKeys } = props;
    this.state = {
      loadedKeys: [],
      loading: true,
      rootAssetNodes: [],
      assetNodesMap: {},
      expandedKeys: AssetTree.toKeys(defaultExpandedKeys || []),
    };
  }

  async componentDidMount() {
    this.loadAssetInfo();
  }

  componentDidUpdate(prevProps: AssetTreeProps) {
    if (!isEqual(prevProps.assetIds, this.props.assetIds)) {
      this.resetExpandedKeys();
      this.loadAssetInfo();
    }
  }

  loadAssetInfo = async () => {
    this.client = this.context(AssetTreeWithTheme.displayName || '')!;
    if (!this.client) {
      console.error(ERROR_NO_SDK_CLIENT);
      return;
    }
    this.setState({ loading: true }, async () => {
      const { assetIds } = this.props;
      let assets: Asset[] = [];
      if (assetIds) {
        if (assetIds.length > 0) {
          assets = await this.client.assets.retrieve(
            assetIds.map(id => ({ id }))
          );
        }
      } else {
        assets = await this.client.assets
          .list({
            filter: { root: true },
            aggregatedProperties: ['childCount'],
          })
          .autoPagingToArray(this.autoPagingToArrayOptions);
      }

      const assetNodesMap = AssetTree.convertAssetsToNodeMap(assets);
      const rootAssetNodes = Object.keys(assetNodesMap).map(Number);

      this.setState({
        rootAssetNodes,
        assetNodesMap,
        loading: false,
      });
    });
  };

  cursorApiRequest = async (assetId: number): Promise<Asset[]> => {
    const result = await this.client.assets
      .list({
        filter: { parentIds: [assetId] },
        aggregatedProperties: ['childCount'],
      })
      .autoPagingToArray(this.autoPagingToArrayOptions);
    if (!result || !Array.isArray(result)) {
      console.error(ERROR_API_UNEXPECTED_RESULTS);
    }
    return result;
  };

  onLoadData = async (treeNode: AntTreeNode) => {
    const eventKey = treeNode.props.eventKey;
    const assetId = eventKey ? Number.parseInt(eventKey, 10) : undefined;
    const { loadedKeys } = this.state;
    const updatedLoadedKeys =
      eventKey && !loadedKeys.includes(eventKey)
        ? [...loadedKeys, eventKey]
        : loadedKeys;

    if (!treeNode.props.children && assetId) {
      const updatedAssetNode = { ...this.state.assetNodesMap[assetId] };
      const assetChildren = await this.cursorApiRequest(assetId);
      const mapUpdate = AssetTree.convertAssetsToNodeMap(assetChildren);

      if (assetChildren.length) {
        assetChildren.sort((a, b) => a.name.localeCompare(b.name));
        updatedAssetNode.children = assetChildren.map(({ id }) => id);
      } else {
        updatedAssetNode.isLeaf = true;
      }

      this.setState(state => ({
        assetNodesMap: {
          ...state.assetNodesMap,
          [assetId]: updatedAssetNode,
          ...mapUpdate,
        },
        loadedKeys: [...updatedLoadedKeys],
      }));
    }
  };

  onSelectNode = (title: string, key?: string) => {
    const assetId = Number.parseInt(key || '', 10);
    const { asset: node, isLeaf } = this.state.assetNodesMap[assetId];
    const { onSelect } = this.props;
    if (onSelect && key) {
      onSelect({ node, key, isLeaf, title });
    }
  };

  onExpand = (expandedKeys: string[]) => {
    this.setState({
      expandedKeys: AssetTree.toKeys(
        expandedKeys.map(key => Number.parseInt(key, 10))
      ),
    });
  };

  renderTreeNode = (nodeIds: number[]) => {
    const { styles } = this.props;
    return nodeIds.map(assetId => {
      const node = this.state.assetNodesMap[assetId];
      const { children, asset, isLeaf } = node;
      return (
        node && (
          <TreeNodeWrapper
            title={this.getDisplayName(asset)}
            key={asset.id}
            isLeaf={isLeaf}
            style={styles && styles.list}
            children={children && this.renderTreeNode(children)}
          />
        )
      );
    });
  };

  resetExpandedKeys = () => {
    this.setState({ expandedKeys: {}, loadedKeys: [] });
  };

  render() {
    const { showLoading } = this.props;
    const { rootAssetNodes, expandedKeys, loading, loadedKeys } = this.state;
    const { onLoadData, onSelectNode, renderTreeNode, onExpand } = this;
    if (showLoading && loading) {
      return <Spin />;
    }
    return (
      <Tree
        loadedKeys={loadedKeys}
        loadData={onLoadData}
        onSelect={(_, e) =>
          onSelectNode(e.node.props.title as string, e.node.props.eventKey)
        }
        expandedKeys={Object.keys(expandedKeys)}
        onExpand={onExpand}
        children={renderTreeNode(rootAssetNodes)}
      />
    );
  }

  private getDisplayName = (asset: Asset) => {
    const { displayName } = this.props;
    if (displayName) {
      return displayName(asset);
    }
    return `${asset.name}${asset.description ? ': ' + asset.description : ''}`;
  };
}

const TreeNodeWrapper = styled(TreeNode)<AntTreeNodeProps>`
  ${({ theme }) => applyThemeFontFamily(theme.gearbox)}
  ${({ theme }) => applyThemeFontSize(theme.gearbox)}
  & span {
    color: ${({ theme }) => theme.gearbox.textColor};
  }
  & .ant-tree-node-selected {
    ${({ theme }) => applyThemeListHighlight(theme.gearbox)}
  }
`;

const AssetTreeWithTheme = withDefaultTheme(AssetTree);
AssetTreeWithTheme.displayName = 'AssetTree';

export { AssetTreeWithTheme as AssetTree };
export { AssetTree as AssetTreeWithoutTheme };
