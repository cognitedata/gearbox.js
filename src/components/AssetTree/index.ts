export * from './AssetTree';
export * from './interfaces';
