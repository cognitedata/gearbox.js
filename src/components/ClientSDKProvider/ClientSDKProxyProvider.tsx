import React, { FC } from 'react';
import { ClientSDKProxyContext } from '../../context/clientSDKProxyContext';
import { wrapInProxies } from '../../proxies/clientSDKProxies';
import { ClientSDKProviderProps } from './interfaces';

export const ClientSDKProxyProvider: FC<ClientSDKProviderProps> = ({
  client,
  children,
}) => (
  <ClientSDKProxyContext.Provider value={wrapInProxies(client)}>
    {children}
  </ClientSDKProxyContext.Provider>
);
