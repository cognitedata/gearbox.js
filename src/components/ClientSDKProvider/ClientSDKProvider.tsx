import React, { FC } from 'react';
import { ClientSDKContext } from '../../context/clientSDKContext';
import { ClientSDKProxyProvider } from './ClientSDKProxyProvider';
import { ClientSDKProviderProps } from './interfaces';

export const ClientSDKProvider: FC<ClientSDKProviderProps> = ({
  client,
  children,
}) => {
  return (
    <ClientSDKContext.Provider value={client}>
      <ClientSDKProxyProvider client={client}>
        {children}
      </ClientSDKProxyProvider>
    </ClientSDKContext.Provider>
  );
};
