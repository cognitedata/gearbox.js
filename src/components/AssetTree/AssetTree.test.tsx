import { Tree } from 'antd';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';
import {
  ASSET_LIST_CHILD,
  ASSET_TREE_STYLES,
  ASSET_ZERO_DEPTH_ARRAY,
  MockCogniteClient,
  sleep,
} from '../../mocks';
import { ClientSDKProvider } from '../ClientSDKProvider';
import { AssetTree } from './AssetTree';

const zeroChild = ASSET_ZERO_DEPTH_ARRAY.findIndex(
  asset => asset.rootId === asset.id
);

class CogniteClient extends MockCogniteClient {
  assets: any = {
    list: jest.fn(),
    retrieve: jest.fn(),
  };
}

const sdk = new CogniteClient({ appId: 'gearbox test' });

configure({ adapter: new Adapter() });

beforeEach(() => {
  sdk.assets.list.mockImplementation(({ filter: { parentIds } }: any) => ({
    autoPagingToArray: async () => {
      if (parentIds) {
        return ASSET_LIST_CHILD.filter(({ parentId }) =>
          parentIds.includes(parentId)
        );
      } else {
        return ASSET_ZERO_DEPTH_ARRAY;
      }
    },
  }));
  sdk.assets.retrieve.mockImplementation((ids: { id: number }[]) => {
    const idsToFind = ids.map(({ id }) => id);
    return ASSET_LIST_CHILD.filter(({ id }) => idsToFind.includes(id));
  });
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('AssetTree', () => {
  it('renders correctly', done => {
    const tree = renderer.create(
      <ClientSDKProvider client={sdk}>
        <AssetTree
          defaultExpandedKeys={[ASSET_ZERO_DEPTH_ARRAY[zeroChild].id]}
        />
      </ClientSDKProvider>
    );
    setImmediate(() => {
      expect(tree).toMatchSnapshot();
      done();
    });
  });

  it('renders correctly with assetIds', done => {
    const tree = renderer.create(
      <ClientSDKProvider client={sdk}>
        <AssetTree assetIds={[ASSET_ZERO_DEPTH_ARRAY[zeroChild].id]} />
      </ClientSDKProvider>
    );
    setImmediate(() => {
      expect(tree).toMatchSnapshot();
      done();
    });
  });

  it('renders correctly when assetIds property reset', async () => {
    const AssetTreeModal = mount(
      <ClientSDKProvider client={sdk}>
        <AssetTree assetIds={[ASSET_ZERO_DEPTH_ARRAY[zeroChild].id]} />
      </ClientSDKProvider>
    );

    await sleep(0);
    AssetTreeModal.update();
    AssetTreeModal.find('.ant-tree-switcher')
      .first()
      .simulate('click');

    await sleep(0);
    AssetTreeModal.update();
    const treeNodesFirstExpanded = AssetTreeModal.find(Tree.TreeNode)
      .map(node => node.props())
      .map(({ eventKey }) => Number(eventKey));
    expect(treeNodesFirstExpanded).toEqual([
      6687602007296940,
      4650652196144007,
    ]);
    expect(sdk.assets.retrieve).toBeCalledWith([{ id: 6687602007296940 }]);
    AssetTreeModal.setProps({
      children: <AssetTree assetIds={undefined} />,
    });

    await sleep(0);
    AssetTreeModal.update();
    AssetTreeModal.setProps({
      children: <AssetTree assetIds={[ASSET_ZERO_DEPTH_ARRAY[zeroChild].id]} />,
    });

    await sleep(0);
    AssetTreeModal.update();
    AssetTreeModal.find('.ant-tree-switcher')
      .first()
      .simulate('click');

    await sleep(0);
    AssetTreeModal.update();
    expect(sdk.assets.list).toBeCalledTimes(3);
    expect(
      AssetTreeModal.find(Tree.TreeNode)
        .map(node => node.props())
        .map(({ eventKey }) => Number(eventKey))
    ).toEqual(treeNodesFirstExpanded);
    expect(sdk.assets.retrieve).toBeCalledTimes(2);
  });

  it('Checks if onSelect is being called', done => {
    const jestTest = jest.fn();
    const AssetTreeModal = mount(
      <ClientSDKProvider client={sdk}>
        <AssetTree onSelect={jestTest} />
      </ClientSDKProvider>
    );
    setImmediate(() => {
      AssetTreeModal.update();
      AssetTreeModal.find('.ant-tree-node-content-wrapper')
        .first()
        .simulate('click');

      expect(jestTest).toBeCalled();
      expect(jestTest.mock.results[0].value).toBe(undefined);
      expect(Object.keys(jestTest.mock.calls[0][0]).sort()).toEqual(
        ['node', 'key', 'isLeaf', 'title'].sort()
      );
      done();
    });
  });

  it("Don't show expand arrows if child count > 0", done => {
    const AssetTreeModal = mount(
      <ClientSDKProvider client={sdk}>
        <AssetTree />
      </ClientSDKProvider>
    );
    setImmediate(() => {
      AssetTreeModal.update();

      expect(
        AssetTreeModal.find(Tree.TreeNode)
          .map(node => node.props())
          .map(({ eventKey, isLeaf }) => ({ id: Number(eventKey), isLeaf }))
      ).toEqual(
        ASSET_ZERO_DEPTH_ARRAY.map(({ id, aggregates }) => ({
          id,
          isLeaf: aggregates ? !aggregates.childCount : false,
        }))
      );

      done();
    });
  });

  it('renders correctly with passed styles prop', done => {
    const wrapper = mount(
      <ClientSDKProvider client={sdk}>
        <AssetTree styles={ASSET_TREE_STYLES} />
      </ClientSDKProvider>
    );
    setImmediate(() => {
      wrapper.update();
      const containerStyle = wrapper
        .find('.ant-tree-treenode-switcher-close')
        .first()
        .prop('style');

      expect(containerStyle === ASSET_TREE_STYLES.list).toBeTruthy();
      done();
    });
  });

  it('renders correctly with after changed the displayName prop', done => {
    const wrapper = mount(
      <ClientSDKProvider client={sdk}>null</ClientSDKProvider>
    );
    wrapper.setProps({
      children: <AssetTree displayName={asset => `${asset.id}`} />,
    });

    const testItemToMatchText = (text: string) => {
      const str = wrapper
        .find('.ant-tree-title')
        .first()
        .text();
      expect(str).toBe(text);
    };

    setImmediate(() => {
      wrapper.update();
      testItemToMatchText('6687602007296940');

      wrapper.setProps({
        children: <AssetTree displayName={asset => `${asset.name}`} />,
      });
      setImmediate(() => {
        wrapper.update();
        testItemToMatchText('Aker BP');
        done();
      });
    });
  });
});
