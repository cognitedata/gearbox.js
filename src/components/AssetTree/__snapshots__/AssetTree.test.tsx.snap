// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetTree renders correctly 1`] = `
<ul
  className="ant-tree ant-tree-icon-hide"
  role="tree"
  unselectable="on"
>
  <li
    className="sc-htpNat dGpXNN ant-tree-treenode-switcher-open"
    role="treeitem"
  >
    <span
      className="ant-tree-switcher ant-tree-switcher_open"
      onClick={[Function]}
    >
      <i
        aria-label="icon: caret-down"
        className="anticon anticon-caret-down ant-tree-switcher-icon"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="caret-down"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
          />
        </svg>
      </i>
    </span>
    <span
      className="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-open"
      onClick={[Function]}
      onContextMenu={[Function]}
      onDoubleClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title="Aker BP: Aker BP"
    >
      <span
        className="ant-tree-title"
      >
        Aker BP: Aker BP
      </span>
    </span>
    <ul
      className="ant-tree-child-tree ant-tree-child-tree-open"
      data-expanded={true}
      role="group"
    >
      <li
        className="sc-htpNat dGpXNN ant-tree-treenode-switcher-close"
        role="treeitem"
      >
        <span
          className="ant-tree-switcher ant-tree-switcher_close"
          onClick={[Function]}
        >
          <i
            aria-label="icon: caret-down"
            className="anticon anticon-caret-down ant-tree-switcher-icon"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="caret-down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
              />
            </svg>
          </i>
        </span>
        <span
          className="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-close"
          onClick={[Function]}
          onContextMenu={[Function]}
          onDoubleClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          title="VAL: Valhall plattform"
        >
          <span
            className="ant-tree-title"
          >
            VAL: Valhall plattform
          </span>
        </span>
      </li>
    </ul>
  </li>
  <li
    className="sc-htpNat dGpXNN"
    role="treeitem"
  >
    <span
      className="ant-tree-switcher ant-tree-switcher-noop"
    />
    <span
      className="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal"
      onClick={[Function]}
      onContextMenu={[Function]}
      onDoubleClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title="VAL: Valhall plattform"
    >
      <span
        className="ant-tree-title"
      >
        VAL: Valhall plattform
      </span>
    </span>
  </li>
</ul>
`;

exports[`AssetTree renders correctly with assetIds 1`] = `
<ul
  className="ant-tree ant-tree-icon-hide"
  role="tree"
  unselectable="on"
>
  <li
    className="sc-htpNat dGpXNN ant-tree-treenode-switcher-close"
    role="treeitem"
  >
    <span
      className="ant-tree-switcher ant-tree-switcher_close"
      onClick={[Function]}
    >
      <i
        aria-label="icon: caret-down"
        className="anticon anticon-caret-down ant-tree-switcher-icon"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="caret-down"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
          />
        </svg>
      </i>
    </span>
    <span
      className="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-close"
      onClick={[Function]}
      onContextMenu={[Function]}
      onDoubleClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title="Aker BP: Aker BP"
    >
      <span
        className="ant-tree-title"
      >
        Aker BP: Aker BP
      </span>
    </span>
  </li>
</ul>
`;
